crates/miden-objects/src/account/component/template/mod.rs

```diff
@@ -110,16 +110,19 @@ impl Deserializable for AccountComponentTemplate {
 /// # use miden_objects::{testing::account_code::CODE, account::{
 /// #     AccountComponent, AccountComponentMetadata, StorageEntry,
 /// #     StorageValueName,
-/// #     AccountComponentTemplate, FeltRepresentation, WordRepresentation},
+/// #     AccountComponentTemplate, FeltRepresentation, WordRepresentation, TemplateType},
 /// #     assembly::Assembler, Felt};
 /// # use miden_objects::account::InitStorageData;
 /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
 /// let first_felt = FeltRepresentation::from(Felt::new(0u64));
 /// let second_felt = FeltRepresentation::from(Felt::new(1u64));
 /// let third_felt = FeltRepresentation::from(Felt::new(2u64));
 /// // Templated element:
-/// let last_element =
-///     FeltRepresentation::new_template("felt", StorageValueName::new("foo")?, None);
+/// let last_element = FeltRepresentation::new_template(
+///     TemplateType::new("felt")?,
+///     StorageValueName::new("foo")?,
+///     None,
+/// );
```

Comment:
I wonder if it would be a nicer API if `new_{template, value}` would not take the optional description and we instead had a `with_description(mut self, description: impl Into<String>) -> Self` method?

Reply:
In terms of API I agree, although it will make parsing a little bit more verbose. Probably a net win though.